File: Pattern18.c
// *********
//  *     *
//   *   *
//    * *
//     *

#include <stdio.h>

void main()
{
    int n;
    printf("Enter the row number:");
    scanf("%d", &n);
    for (int i = 1; i <= n; i++)
    {
        for (int k = 1; k <= i; k++)
        {
            printf(" ");
        }
        int col = i == 1 ? 2 * n - 1 : n;
        for (int j = i; j <= col; j++)
        {
            if (col < 2 * n - 1)
            {
                if (j == i || j == n)
                {
                    printf("* ");
                }
                else
                {
                    printf("  ");
                }
            }
            else
            {
                printf("*");
            }
        }
        printf("\n");
    }
}

File: Pattern18.go
// *********
//  *     *
//   *   *
//    * *
//     *

package main

import "fmt"

func main() {
	var n int
	var col int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d", &n)

	for i:=1;i<=n;i++{
		for k:=1;k<=i;k++{
			fmt.Print(" ")
		}
		if i==1{
			col = 2*n-1
		}else{
			col = n
		}
		for j:=i;j<=col;j++{
			if col<2*n-1{
				if(j==i || j==n){
					fmt.Print("* ")
				}else{
					fmt.Print("  ")
				}
			}else{
				fmt.Print("*")
			}
	
		}
		fmt.Println()
	}
}
